internal/services/devcenter/dev_center_project_pool_resource.go

```diff
@@ -126,14 +138,21 @@ func (r DevCenterProjectPoolResource) Create() sdk.ResourceFunc {
 			parameters := pools.Pool{
 				Location: location.Normalize(model.Location),
 				Properties: &pools.PoolProperties{
-					DevBoxDefinitionName:  pointer.To(model.DevBoxDefinitionName),
-					NetworkConnectionName: pointer.To(model.DevCenterAttachedNetworkName),
-					LicenseType:           pointer.To(pools.LicenseTypeWindowsClient),
-					StopOnDisconnect:      expandDevCenterProjectPoolStopOnDisconnect(model.StopOnDisconnectGracePeriodMinutes),
+					DevBoxDefinitionName:         pointer.To(model.DevBoxDefinitionName),
+					NetworkConnectionName:        pointer.To(model.DevCenterAttachedNetworkName),
+					LicenseType:                  pointer.To(pools.LicenseTypeWindowsClient),
+					ManagedVirtualNetworkRegions: expandDevCenterProjectManagedVirtualNetworkRegions(model.ManagedVirtualNetworkRegions),
+					StopOnDisconnect:             expandDevCenterProjectPoolStopOnDisconnect(model.StopOnDisconnectGracePeriodMinutes),
 				},
 				Tags: pointer.To(model.Tags),
 			}
 
+			if len(model.ManagedVirtualNetworkRegions) != 0 {
+				parameters.Properties.VirtualNetworkType = pointer.To(pools.VirtualNetworkTypeManaged)
+			} else {
+				parameters.Properties.VirtualNetworkType = pointer.To(pools.VirtualNetworkTypeUnmanaged)
+			}
```

Comment:
Nit: we generally try to avoid `else` statements where possible and instead just set the content from the `else` block initially, then update based on whether the `if ...` conditional passes

```suggestion
			parameters.Properties.VirtualNetworkType = pointer.To(pools.VirtualNetworkTypeUnmanaged)
			if len(model.ManagedVirtualNetworkRegions) != 0 {
				parameters.Properties.VirtualNetworkType = pointer.To(pools.VirtualNetworkTypeManaged)
			}
```